pkg/apis/build/v1alpha1/source.go

```diff
@@ -8,11 +8,24 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// Container describes the source code bundle container to pull
+type Container struct {
+	// Image reference, i.e. quay.io/org/image:tag
+	Image string `json:"image"`
+}
+
 // Source describes the Git source repository to fetch.
 type Source struct {
 	// URL describes the URL of the Git repository.
+	//
+	// +optional
 	URL string `json:"url"`
 
+	// Container
+	//
+	// +optional
+	Container *Container `json:"container"`
+
```

Comment:
a confirmation / agreement comment here @HeavyWombat @SaschaSchwarze0 @otaviof  ... having  pointer ref here is conducive to allowing for multiple "implementation" choices for the upload

i.e. we add a new pointer that enables use of a `kubectl cp` equivalent

so cool deal

we probably will just need to make sure when we get to that point that there is validation that only allows one of these optional fields related to upload mechanism to be set

Reply:
Good point. I have not thought about it much, I guess I kind of was under the impression that every "type" of input would eventually end up having their respective individual entry here, therefore making it a pointer reference. Depending on how many other "types" we want to introduce, we could end up with a lot of optional fields, but I guess this is ok.

Reply:
yeah we have had as many a 4 "options" in some places in openshift ... so yes "seems OK"

bottom line, we are all good here :-) .... occasionally it is just good to provide positive feedback in PRs as well :-)